MarkBind/markbind
# Patch Glyphicons and Add Font Awesome Icons

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Documentation update
• [X] Enhancement to an existing feature
• [X] Other, migration


Part of #298 and MarkBind/vue-strap#53



**What is the rationale for this request?**
`Glyphicons` has been deprecated from Bootstrap 4.x assets, which caused missing icons within MarkBind's components.

**What changes did you make? (Give an overview)**
- Import existing `Glyphicons` CSS file from [this repository](https://github.com/ohpyupi/glyphicons-only-bootstrap).
- Added `Font Awesome`'s set of free icons into our assets
- Reference both icons' style sheet in `page.ejs`
- Add `Font Awesome` to `userDefinedVariables` map, same as `Glyphicons`
- Update user guide in [Font Awesome](https://deploy-preview-301--markbind.netlify.com/userguide/contentauthoring#font-awesome-icons)

Discussion |
--- |
Font Awesome was added as a (better) replacement to Glyphicons, as there is no license that explicitly mentions the use of Glyphicons in Bootstrap 4.x (only mentioned dropping it). Further investigation is required, and if found that it violates Glyphicon's terms of use, we will remove it from our application. | 
**Part Two** After team discussion, we decided that importing Glyphicon's CSS file from Bootstrap 3.x is within fair use. Thus we will continue supporting Glyphicons in MarkBind. Acknowledgements of any other projects or assets we use will be done in our repository's README page.

**Is there anything you'd like reviewers to focus on?**
- New `getIconsMap` method, to collate all icon assets.
- Phrasing of user guide 

**Testing instructions:**
- Ensure you have Bootstrap's 4.x `.min.js` file in your asset folder
- Check that Glyphicons and Font Awesome icons work by using `{{glyphicon_home}}` and `{{fab_playstation}}` in any `index.md` before serving your site.
